openid_connect.js

```diff
@@ -253,6 +250,16 @@ function validateIdToken(r) {
             r.error("OIDC ID Token validation error: nonce from token (" + r.variables.jwt_claim_nonce + ") does not match client (" + client_nonce_hash + ")");
             validToken = false;
         }
+    } else {
+        var newSession = (!r.variables.refresh_token || r.variables.refresh_token == "-");
+        if (newSession) {
+            r.error("OIDC ID Token validation error: nonce absent in token");
+            validToken = false;
+        } else {
+            // Tolerate missing nonce on renewals
+            // https://github.com/nginxinc/nginx-openid-connect/pull/104#issuecomment-2433361196
+            r.warn("OIDC ID Token validation error: No nonce claim");
```

Comment:
Text says error but it's only a warn.
```suggestion
            r.warn("OIDC ID Token validation did not include nonce");
```

Reply:
@lcrilly I added an additional commit with revisions that I believe cover this suggestion, between the time you started reviewing and when you posted this comment. Sorry about that.

Reply:
@ag-TJNII Don't worry about that, all good - and thanks for contributing, and more specifically catching that we should be validating nonce in the refresh case (if available), that was the important part :)

Reply:
Do we really need this warning considering that the presence of a nonce for refresh cases is not mandatory? Essentially, during a refresh, the UserAgent isn’t involved in the process, and replay attacks aren’t possible in this case.

Reply:
@route443 Ah, I see - it's in the [Successful Refresh Token section of the docs](https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokenResponse) not in the ID token section...

`it SHOULD NOT have a nonce Claim, even when the ID Token issued at the time of the original authentication contained nonce; however, if it is present, its value MUST be the same as in the ID Token issued at the time of the original authentication` 

It was ambiguous until [recently](https://bitbucket.org/openid/connect/pull-requests/341/diff).

So yes, by the language of the above we should probably just remove the else {}

Reply:
@route443 Changes now reflected in the latest version.